src/app/services/validators/dataset_validator_service.py

```diff
@@ -136,3 +183,18 @@ def _missing_keys_response(missing_keys: list[str]) -> tuple[bool, str]:
         """
 
         return False, ", ".join(missing_keys)
+
+    @staticmethod
+    def _publish_dataset_error_to_topic(
+        error_message: DatasetError,
+    ) -> None:
+        """
+        Publishes dataset error response to google pubsub topic.
+
+        Parameters:
+        error_message (str): error message to be published.
+        """
+
+        dataset_repository = DatasetFirebaseRepository()
+        dataset_writer_service = DatasetWriterService(dataset_repository)
+        dataset_writer_service.try_publish_dataset_error_to_topic(error_message)
```

Comment:
Just a thought on code refactoring. I am thinking it will be less convoluted if we just put `try_publish_dataset_error_to_topic` from `dataset_writer_service` to here given that the class is meant to handle the part that start writing dataset to database, calling both the `DatasetFireBaseRepository` and `DatasetWriterService` class in this validator seems not quite logical.